runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/config/CompressionClientConfig.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package aws.smithy.kotlin.runtime.http.config
```

Comment:
**correctness/discussion**: Now that `CompressionAlgorithm` isn't tied to HTTP should this configuration type live in `smithy-client` instead?

Reply:
I think that makes sense

Reply:
Under `aws.smithy.kotlin.runtime.client.config` or just `aws.smithy.kotlin.runtime.client` I would guess `config` but just double checking

Reply:
`config` is probably more correct but it would seem we weren't consistent at first glance